packages/motherduck/src/Adaptor.js

```diff
@@ -0,0 +1,673 @@
+    } catch (error) {
+      console.error('Delete operation failed:', error);
+      throw error;
+    }
+  };
+}
+
+/**
+ * Create a table in the database
+ * @public
+ * @example
+ * createTable('users', [
+ *   { name: 'id', type: 'INTEGER PRIMARY KEY' },
+ *   { name: 'name', type: 'VARCHAR(100)', required: true },
+ *   { name: 'age', type: 'INTEGER' }
+ * ])
+ * @function
+ * @param {string} tableName - The name of the table to create
+ * @param {array} columns - Array of column definitions
+ * @param {object} [options] - Optional options argument
+ * @returns {Operation}
+ */
+export function createTable(tableName, columns, options = {}) {
+  return state => {
+    try {
+      // Validate table name for security
+      validateSqlIdentifier(tableName);
+      
+      if (!columns || columns.length === 0) {
+        console.log('No columns provided; skipping table creation.');
+        const nextState = { ...state, data: 'No columns provided for table creation' };
+        return nextState;
+      }
+      
+      const columnDefinitions = columns.map(col => {
+        let definition = `${col.name} ${col.type}`;
+        if (col.required) definition += ' NOT NULL';
+        if (col.unique) definition += ' UNIQUE';
+        if (col.default !== undefined) {
+          if (typeof col.default === 'string') {
+            definition += ` DEFAULT '${col.default}'`;
+          } else {
+            definition += ` DEFAULT ${col.default}`;
+          }
+        }
+        return definition;
+      }).join(', ');
+      
+      const sqlQuery = `CREATE TABLE ${tableName} (${columnDefinitions})`;
+      
+      console.log('Preparing to create table:', tableName);
+      return queryHandler(state, sqlQuery, options);
+    } catch (error) {
+      console.error('Create table operation failed:', error);
+      throw error;
+    }
+  };
+}
+
+/**
+ * Drop a table from the database
+ * @public
+ * @example
+ * dropTable('old_users')
+ * @function
+ * @param {string} tableName - The name of the table to drop
+ * @param {object} [options] - Optional options argument
+ * @returns {Operation}
+ */
+export function dropTable(tableName, options = {}) {
+  return state => {
+    try {
+      // Validate table name for security
+      validateSqlIdentifier(tableName);
+      
+      const ifExists = options?.ifExists ? 'IF EXISTS ' : '';
+      const sqlQuery = `DROP TABLE ${ifExists}${tableName}`;
+      
+      console.log('Preparing to drop table:', tableName);
+      return queryHandler(state, sqlQuery, options);
+    } catch (error) {
+      console.error('Drop table operation failed:', error);
+      throw error;
+    }
+  };
+}
+
+/**
+ * Describe a table structure
+ * @public
+ * @example
+ * describeTable('users')
+ * @function
+ * @param {string} tableName - The name of the table to describe
+ * @param {object} [options] - Optional options argument
+ * @returns {Operation}
+ */
+export function describeTable(tableName, options = {}) {
```

Comment:
Since you have a generic `query` function, I'm going to ask: what is the value of some of these smaller functions? Are you adding them because you need them for your use-case, or just because they're useful to users?

Our usual approach to adaptors is to provide one low-level function which can do basically anything - in this case, `query()`. And then provide higher-order  functions either where they add value, or where users are asking for them specifically.

In the case of `describe` - how does the result get written back to state? Is it a JSON object? Does it make more sense to just log a string?

I don't want to push this too far, but every function has a cost (testing, maintenance, documentation, usability). Do we think the cost of this function is worthwhile?